packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Snippets/Snippet.cs

```diff
@@ -50,6 +50,9 @@ public static partial class Snippet
         public static ValueExpression ThrowExpression(ValueExpression expression) => new KeywordExpression("throw", expression);
 
         public static ValueExpression NullCoalescing(ValueExpression left, ValueExpression right) => new BinaryOperatorExpression("??", left, right);
+
+        public static ScopedApi<T> NullConditional<T>(this ScopedApi<T> valueExpression, bool nullConditional = false) => nullConditional ? new NullConditionalExpression(valueExpression.Original).As<T>() : valueExpression;
```

Comment:
If we only have this we could only use the `NullConditional` on `ScopedApi<T>`, but we should also be able to call it on `ValueExpression`, maybe we could add another extension method for `ValueExpression` to catch up all the remaining cases.

Reply:
I think the default value here of the flag should be `true`. It does not make sense if we call it and nothing happens like this:
```
foo.NullConditional()
```
this is very misleading.